### PR TITLE
feat(plugins): Web/Vesktop AI Noise Suppression

### PR DESCRIPTION
**Pull request by Vap0r1ze**
still needs to be tested for long term performance
image
image
image

### CODE REVIEW

File: src/plugins/rnnoise.vencordDesktop/icons.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+export const SupressionIcon = ({ enabled }: { enabled: boolean; }) => enabled
```
Comment (Vendicated) [inline src/plugins/rnnoise.vencordDesktop/icons.tsx:19]: this is very insane, do you think we can fetch it from discord?

File: src/plugins/rnnoise.vencordDesktop/index.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+            Enable AI noise suppression! Make some noise&mdash;like becoming an air conditioner, or a laptop fan&mdash;while speaking. Your friends will hear nothing but your beautiful voice ✨
+        </div>
+    </div>;
+}
+
+export default definePlugin({
+    name: "AI Noise Suppression",
+    description: "Uses an open-source AI model (RNNoise) to remove background noise from your microphone",
+    authors: [Devs.Vap],
+    settings,
```
Comment (Vendicated) [inline src/plugins/rnnoise.vencordDesktop/index.tsx:77]:
```suggestion
    settings,
    enabledByDefault: true,
```

File: src/plugins/rnnoise.vencordDesktop/index.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+        // Cleanup
+        const onEnded = () => {
+            stream.removeEventListener("inactive", onEnded);
+            rnnoise.disconnect();
+            source.disconnect();
+            audioCtx.close();
+            rnnoise.destroy();
+        };
+        stream.addEventListener("inactive", onEnded);
```
Comment (Vendicated) [inline src/plugins/rnnoise.vencordDesktop/index.tsx:137]:
```suggestion
        stream.addEventListener("inactive", onEnded, { once: true });
```
then u don't need to remove it

Comment (Vap0r1ze) [reply src/plugins/rnnoise.vencordDesktop/index.tsx:138]: ur so cool

File: src/plugins/rnnoise.vencordDesktop/icons.tsx (hunk 1 above)
Comment (Vap0r1ze) [reply src/plugins/rnnoise.vencordDesktop/icons.tsx:19]: uhhhhhhhhhhhhhhhhhhh maybe

Comment (Vap0r1ze) [reply src/plugins/rnnoise.vencordDesktop/icons.tsx:19]:
```js
function Dse(e) {
    var t = e.width
        , n = void 0 === t ? 24 : t
        , o = e.height
        , i = void 0 === o ? 24 : o
        , a = e.color
        , s = void 0 === a ? "currentColor" : a
        , l = e.foreground
        , c = xse(e, ["width", "height", "color", "foreground"]);
    return (0,
    r.jsxs)("svg", Lse(function(e) {
        for (var t = 1; t < arguments.length; t++) {
            var n = null != arguments[t] ? arguments[t] : {}
                , r = Object.keys(n);
            "function" == typeof Object.getOwnPropertySymbols && (r = r.concat(Object.getOwnPropertySymbols(n).filter((function(e) {
                return Object.getOwnPropertyDescriptor(n, e).enumerable
            }
            ))));
            r.forEach((function(t) {
                wse(e, t, n[t])
            }
            ))
        }
        return e
    }({}, (0,
    kg.Z)(c)), {
        width: n,
        height: i,
        viewBox: "0 0 24 24",
        xmlns: "http://www.w3.org/2000/svg",
        children: [(0,
        r.jsx)("path", {
            className: l,
            d: "M10.889 4C10.889 3.44772 11.3367 3 11.889 3H12.1112C12.6635 3 13.1112 3.44772 13.1112 4V20C13.1112 20.5523 12.6635 21 12.1112 21H11.889C11.3367 21 10.889 20.5523 10.889 20V4Z",
            fill: s
        }), (0,
        r.jsx)("path", {
            className: l,
            d: "M6.44439 6.25C6.44439 5.69772 6.89211 5.25 7.44439 5.25H7.66661C8.2189 5.25 8.66661 5.69772 8.66661 6.25V17.75C8.66661 18.3023 8.2189 18.75 7.66661 18.75H7.44439C6.89211 18.75 6.44439 18.3023 6.44439 17.75V6.25Z",
            fill: s
        }), (0,
        r.jsx)("path", {
            className: l,
            d: "M3.22222 15.375C3.77451 15.375 4.22222 14.9273 4.22222 14.375L4.22222 9.625C4.22222 9.07272 3.77451 8.625 3.22222 8.625H3C2.44772 8.625 2 9.07272 2 9.625V14.375C2 14.9273 2.44772 15.375 3 15.375H3.22222Z",
            fill: s
        }), (0,
        r.jsx)("path", {
            className: l,
            d: "M22.0001 13.25C22.0001 13.8023 21.5523 14.25 21.0001 14.25H20.7778C20.2255 14.25 19.7778 13.8023 19.7778 13.25V10.75C19.7778 10.1977 20.2255 9.75 20.7778 9.75H21.0001C21.5523 9.75 22.0001 10.1977 22.0001 10.75V13.25Z",
            fill: s
        }), (0,
        r.jsx)("path", {
            className: l,
            d: "M16.3333 7.5C15.781 7.5 15.3333 7.94772 15.3333 8.5V15.5C15.3333 16.0523 15.781 16.5 16.3333 16.5H16.5555C17.1078 16.5 17.5555 16.0523 17.5555 15.5V8.5C17.5555 7.94772 17.1078 7.5 16.5555 7.5H16.3333Z",
            fill: s
        })]
    }))
}```

**Comment by Vap0r1ze** [comment]
This PR has been in limbo cus my friend told me my mic was crackly with this plugin and not with discord desktop, what I didnt realize is that my mic had insane feedback from something and Krisp is just really good at removing it.

So fixed my mic and sat in a call with myself for like an hour singing and playing a cpu intensive game and it was fine so imma assume this pr is working

**Comment by Vap0r1ze** [comment]
can i get based check ?? ?? ? !??

**Comment by Vap0r1ze** [comment]
can i get based check ?? ?? ? !??

**Comment by Vendicated** [comment]
no ❤️

File: src/plugins/rnnoise.vencordDesktop/index.tsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+function NoiseSupressionPopout() {
+    const { isEnabled } = settings.use();
+
+    return <div className={cl("popout")}>
+        <div className={cl("popout-heading")}>
+            <span>Noise Supression</span>
+            <Switch checked={isEnabled} onChange={setEnabled} />
+        </div>
+        <div className={cl("popout-desc")}>
+            Enable AI noise suppression! Make some noise&mdash;like becoming an air conditioner, or a laptop fan&mdash;while speaking. Your friends will hear nothing but your beautiful voice ✨
```
Comment (Vendicated) [inline src/plugins/rnnoise.vencordDesktop/index.tsx:69]:
```suggestion
            Enable AI noise suppression! Make some noise&mdash;like becoming an air conditioner, or a vending machine fan&mdash;while speaking. Your friends will hear nothing but your beautiful voice ✨
```

File: src/preload.ts [hunk 5]
```diff
@@ -25,10 +25,12 @@ import VencordNative from "./VencordNative";
 
 contextBridge.exposeInMainWorld("VencordNative", VencordNative);
 
+const isVesktopOrInsane = !process.env.DISCORD_PRELOAD;
+
 // Discord
 if (location.protocol !== "data:") {
     // #region cssInsert
-    const rendererCss = join(__dirname, "renderer.css");
+    const rendererCss = join(__dirname, isVesktopOrInsane ? "vencordDesktopRenderer.css" : "renderer.css");
```
Comment (Vendicated) [inline src/preload.ts:33]: wait does it even download this

Comment (Vap0r1ze) [reply src/preload.ts:33]: cries

**Comment by imlighty** [comment]
Any updates on this?

**Comment by Vendicated** [comment]
i forgor anyway it's probably mergeable

**Comment by imlighty** [comment]
Alright

**Comment by imlighty** [comment]
For a moment I thought it was already merged

**Review by Vendicated — APPROVED** [review]

**Comment by Vendicated** [comment]
too lazy to read new changes they are proabbly fire :fire:

**Pull request merged**